### PR TITLE
Individual community section pages

### PR DESCRIPTION
**Pull request by tusharjain0022**
### Solves Issue #158 
<Do not delete this template>
Description:
 
I've created a reusable react component `CommunityTemplate.js` that will take the community logo, heading, and details as a prop, and render it accordingly to all the community pages. 

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes which were demanded in the linked issue.
- [x] I've tested my code on a Chromium based browser.
- [x] I've tested my code on Mozilla Firefox.
- [x] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial
<provide a relevant screenshot or any testimonial for the solved issue if applicable>

Screenshot (482)

### CODE REVIEW

**Comment by SaravgiYash** [comment]
@tusharjain0022 when I click on the community page I see this:
Capture

then I need to scroll up to see the image and text

**Comment by ankiiitraj** [comment]
Alternate approach:
Make an array and every element of this array will contain the required details such as imgPath, description, title, members, and on the redirection we can render those details which URL is like if it is /cp, we can use CP object from the array.
This will require a single community component

**Comment by ankiiitraj** [comment]
> @tusharjain0022 when I click on the community page I see this:
> Capture
> 
> then I need to scroll up to see the image and text

This is happening because react-router-dom remembers the scroll position of the previous page.
I've got a solution to this I think.
@Yashs911 go ahead a create an issue for this!

File: client/src/components/Community/Communities/CompetitiveCoding.community.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+import { CommunityDetails as community } from "./CommunityDetails";
 
 function CompetitiveCoding() {
   return (
     <div className="competitive-coding">
-      <h1>This is Competitive coding community page.</h1>
+      <CommunityTemplate Community_details={community.CP} />
       <CompetitiveCodingMembers />
     </div>
   );
```
Comment (anktrj) [inline client/src/components/Community/Communities/CompetitiveCoding.community.js]: These community components are also not required, Just keep one generic community component and render details based on URL.
Thoughts?

**Review by ankiiitraj — CHANGES_REQUESTED** [review]

File: client/src/components/Community/Communities/CompetitiveCoding.community.js (hunk 1 above)
Comment (tusharjain0022) [reply client/src/components/Community/Communities/CompetitiveCoding.community.js]: Ok now I got it, it would make the file structure more clean and easy to maintain/edit 😃.So we're targeting towards replacing different community/member files with a single file that will render details according to the URL.  It would require changing the routes for the different community and member files and also we'll need to restructure the Member and Community folder also 🤔 . Correct me if I'm wrong. @ankiiitraj @EmperorYP7

Comment (anktrj) [reply client/src/components/Community/Communities/CompetitiveCoding.community.js]: So, What I'm thinking now is having 1 component and two array files one for community details and another for member details and then we can use these arr files for data in both the components of members and community.

Yup this will require some file structure changes and if possible we should go for this!
inside the community, we can have routes like /community/CP

Comment (tusharjain0022) [reply client/src/components/Community/Communities/CompetitiveCoding.community.js]: I'll try to implement this in my next commit. 🙌

Comment (EmperorYP7) [reply client/src/components/Community/Communities/CompetitiveCoding.community.js]: Well, on the member's page we'll be rendering every community's members. File structure needs to be revised. Try to implement this for competitive coding for now @tusharjain0022. For other communities, fill in some dummy text.

File: client/src/components/Community/Communities/CommunityTemplate.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
+import "./CommunityTemplate.css";
+
+function CommunityTemplate(props) {
```
Comment (EmperorYP7) [inline client/src/components/Community/Communities/CommunityTemplate.js]: This component can be used as a generic template for individual community component.

File: client/src/components/Community/Communities/CompetitiveCoding.community.js [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
+import "bootstrap/dist/css/bootstrap.min.css";
+import CompetitiveCodingMembers from "../../Members/Community/CompetitiveCoding.members";
+import CommunityTemplate from "./CommunityTemplate";
+import { CommunityDetails as community } from "./CommunityDetails";
 
 function CompetitiveCoding() {
   return (
     <div className="competitive-coding">
-      <h1>This is Competitive coding community page.</h1>
+      <CommunityTemplate Community_details={community.CP} />
```
Comment (EmperorYP7) [inline client/src/components/Community/Communities/CompetitiveCoding.community.js]: Great!

**Review by EmperorYP7 — COMMENTED** [review]
Nicely done @tusharjain0022!

File: client/src/components/Community/Communities/CompetitiveCoding.community.js (hunk 1 above)
Comment (tusharjain0022) [reply client/src/components/Community/Communities/CompetitiveCoding.community.js]: I've implemented this on the community section only, Now we've only one file `CommunityTemplate.js` that will render the details depending upon the URL, Same can be implemented for the Member section also. For now, I've moved all the unused files to the `to_be_removed` folder.

File: client/src/components/Community/CommunityList.js [hunk 4]
```diff
@@ -1,13 +1,13 @@
...
+        linkName: "competitive_coding",
         name: "Competitive Coding",
         description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis, lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo, vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis"
     },
     {
         id: 1,
-        linkName: "electronics-iot",
+        linkName: "electronics_iot",
         name: "Electronics and IoT",
         description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis, lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo, vel fringilla est ullamcorper eget nulla facilisi etiam dignissim diam quis"
```
Comment (anktrj) [inline client/src/components/Community/CommunityList.js:12]: Is this file required?

File: client/src/components/Community/Communities/CommunityTemplate.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import "bootstrap/dist/css/bootstrap.min.css";
+import "./CommunityTemplate.css";
+import { CommunityDetails as community } from "./CommunityDetails";
+// import CompetitiveCodingMembers from "../../Members/Community/CompetitiveCoding.members";
```
Comment (anktrj) [inline client/src/components/Community/Communities/CommunityTemplate.js]: Cleanup required

File: client/src/App.js [hunk 6]
```diff
@@ -9,14 +9,15 @@ import Footer from "./components/Footer/Footer";
...
 import Navbar from "./components/Navbar/Navbar";
-import CompetitiveCoding from './components/Community/Communities/CompetitiveCoding.community';
-import Design from './components/Community/Communities/Design.community';
-import Electronics from './components/Community/Communities/Electronics.community';
-import MachineLearning from './components/Community/Communities/MachineLearning.community';
-import Robotics from './components/Community/Communities/Robotics.community';
-import WebDAppD from './components/Community/Communities/WebDAppD.community';
-import Admin from './components/Admin/Admin';
-import ProtectedRoute from "./components/ProtectedRoute/ProtectedRoute"
+// import CompetitiveCoding from "./components/Community/Communities/CompetitiveCoding.community";
```
Comment (anktrj) [inline client/src/App.js]: Cleanup required

**Review by ankiiitraj — CHANGES_REQUESTED** [review]

File: client/src/components/Community/CommunityList.js (hunk 4 above)
Comment (EmperorYP7) [reply client/src/components/Community/CommunityList.js:12]: Yes it does.. just for simplicity.

File: client/src/App.js (hunk 6 above)
Comment (tusharjain0022) [reply client/src/App.js]: Done !!

File: client/src/components/Community/Communities/CommunityTemplate.js (hunk 5 above)
Comment (tusharjain0022) [reply client/src/components/Community/Communities/CommunityTemplate.js]: Done !!

**Review by ankiiitraj — APPROVED** [review]
